sw/device/lib/dif/dif_spi_host.c

```diff
@@ -0,0 +1,383 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_spi_host.h"
+
+#include <assert.h>
+#include <stddef.h>
+
+//#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/runtime/hart.h"
+//#include "sw/device/lib/runtime/log.h"
+
+#include "spi_host_regs.h"  // Generated.
+
+// We create weak symbol aliases for the FIFO write and read functions so the
+// unit tests can provide mocks.  The mocks provide for separate testing of
+// the FIFO functions and the overall transaction management functions.
+void spi_host_fifo_write_alias(const dif_spi_host_t *spi_host, const void *src,
+                               uint16_t len)
+    OT_ATTR_ALIAS("dif_spi_host_fifo_write");
```

Comment:
Do these need to be after the function? I think in general we put aliases before when possible. (Compare `#[...]` in Rust.)

Reply:
They also work before.  The GCC documentation shows them as after, which is why I did it that way.